bot.js

```diff
@@ -335,36 +343,111 @@ bot.isChannel = function(name) {
+  //   lines: 2, // number of lines to spam
+  //   replaceNewlines: false, // whether to replace newlines with | while figuring out if it'll spam
+  //   pmExtra: false, // whether to PM the whole message if it overflows
+  // }
+  var customReply = function(opts, ...args) {
+    // Note, this value is based on what our client does (https://github.com/martynsmith/node-irc/blob/e4000b7a8ac42d9eb16fb6c3f362e1425d664f4b/lib/irc.js#L1069), which may differ from the reality of what a given irc server enforces.
+    //
+    let maxLineChars = Math.min(bot.client.maxLineLength - to.length, 
+                                bot.client.opt.messageSplit);
+
+    // default opts
+    opts = Object.assign({
+      trim: true,
+      lines: 2,
+      replaceNewlines: false,
+      pmExtra: false,
+    }, opts);
+
+    // If it's a pm, all options get ignored
+    if(isPm) {
+      spamReply.apply(this, args);
+      return;
+    }
+
+    // Since this is a message to a channel, we're going to now apply 'custom'
+    // logic to it.
+    // First, get the string representation which the module requested to send.
+    let repStr = bot.stringifyArgs.apply(this, args);
+    // Trim it if requested
+    if(opts.trim) {
+      repStr = repStr.trim();
+    }
+    // Figure out the number of lines this would span, taking into account too-long lines splitting across lines.
+    let lines = repStr.split("\n");
+    let numLines = lines.reduce((sum, line) => {
+      // how many lines does this consume after splitting over maxLineChars?
+      let linesSplitOver = Math.ceil(line.length / maxLineChars);
+      // even a 0 length line gets printed and consumes space
+      return sum + Math.max(1, linesSplitOver);
+    }, 0);
+
+    // If it fits in the max lines allowed, we don't have to modify anything
+    if(numLines <= opts.lines) {
+      // Cool, we can just say it raw and be done with it.
+      bot.client.say(to, repStr);
+      return;
+    }
+
+    // If replaceNewlines is set, we see if ignoring '\n' and joining stuff with '|' fixes it.
+    if(opts.replaceNewlines) {
+      let withoutNewlines = repStr.replace(/\n/g, " | ");
+      if(withoutNewlines.length <= (maxLineChars * opts.lines)) {
+        // Stripping was enough, we're done
+        bot.client.say(to, withoutNewlines);
+        return;
+      }
 
-    if(bot.isChannel(chan)) {
-      bot.callModuleFn('channotice', [bot.client.nick, to, repStr]);
+      // This isn't fitting, say a stripped version
+      bot.client.say(to, withoutNewlines.substring(0, maxLineChars - 4) + " ...");
+      // don't return, we might have to spit out extra
     } else {
-      bot.callModuleFn('pmnotice', [bot.client.nick, to, repStr]);
+      // we know it doesn't fit like this already, let's say a trimmed down version.
+      let linesToSay = [];
+      let maxLength = maxLineChars * opts.lines - 4; // -4 to leave room for '...'
+      for(let i=0; i < lines.length; i++) {
```

Comment:
Ahh, I see. Maybe a `map()` +`take(opts.lines)`?

(Really, though, this is just a nice bikeshed. If you're happy with it, I'm happy with it =D)